lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -1,18 +1,44 @@
 'use strict';
 
-var fs = require('graceful-fs');
+var path = require('path');
 
 var fo = require('../../file-operations');
 
 function writeSymbolicLink(file, optResolver, onWritten) {
-  // TODO handle symlinks properly
-  fs.symlink(file.symlink, file.path, function(symlinkErr) {
-    if (fo.isFatalOverwriteError(symlinkErr)) {
-      return onWritten(symlinkErr);
-    }
-
-    onWritten();
-  });
+  var srcPath = file.symlink;
+
+  var isDirectory = file.isDirectory();
```

Comment:
The least invasive -- but pretty ugly -- way I see to handle both our examples would be to proceed with `isDirectory()` as you were, but if that returns false to do another `fs.stat`. E.g. something like

```js
  var isDirectory = file.isDirectory();
  if (!isDirectory && file.isSymbolic()) {
    isDirectory = fs.statSync(file.symlink).isDirectory();
  }
```
or somehow making that an async fs call.

Of course this is only ever an issue on Windows (the `type` arg to `fs.symlink` is ignored otherwise) and we'll want to avoid this complication entirely when not on Windows.

At one point I was thinking it might make sense to carry both regular and symbolic stats on symbolic vinyls. That would've made the recent issue with symlink resolution (the roundabout recursion which we managed to excise) a little less opaque, for example.

It'd be confusing to users to have two stats on vinyls though, so I guess it might only help within the bounds of vinyl-fs streams.

But if we don't keep both, notice that if we do indeed need an extra `fs.stat` here, it might be the _fourth_  stat/lstat shuffle for the same file in some scenarios: node-glob does an `lstat` and then a `stat`, we then do an `lstat` if resolveSymlinks is false in vfs.src() only to then follow it up here with another `stat` to implement junctions properly!